gcp_deepvariant_runner.py

```diff
@@ -669,10 +669,18 @@ def _validate_and_complete_args(pipeline_args):
     raise ValueError(
         '--job_name_prefix must meet GCP label restrictions: {}'.format(
             pipeline_args.job_name_prefix))
-  if pipeline_args.preemptible and pipeline_args.max_preemptible_tries <= 0:
-    raise ValueError('--max_preemptible_tries must be greater than zero.')
-  if pipeline_args.max_non_preemptible_tries <= 0:
-    raise ValueError('--max_non_preemptible_tries must be greater than zero.')
+  if pipeline_args.preemptible:
```

Comment:
Doesn't this change the behaviour so that the user has to provide a value for the attempt count?

I would have thought you would just change the code to ignore the uninteresting parameter and leave the defaults unchanged

Reply:
Correct, now users must set either `--max_non_preemptible_tries` or [`--preemptible`, `--max_preemptible_tries`] pair.

First I wanted to do what you mentioned, ignore `--max_non_preemptible_tries` value when `--preemptible` is set. However, I decided not to do that and instead explicitly make the user to choose a retry value that reflects their intention. I decided to do this because this parameter can significantly affect the overall cost of running DV.

Note these flag validation tests are happening before starting actual DV stages. So user will get this error (almost) immediately for (almost) $0 cost.

Reply:
I still think it makes more sense to basically have a --attempts flag and --preemptible flag, no?

Reply:
I modify our code and replaced `--max_non_preemptible_tries` and `--max_preemptible_tries` with `--attempts`. Note, we need to update our docs after this change is released:
    https://cloud.google.com/genomics/docs/tutorials/deepvariant

Reply:
Will this break existing users?  I suspect so... In general we would need to keep support for the old flags for at least a release

Reply:
Yes it does break existing users, however, it will cause a clear error message at the first stage of launching DV runner (before starting more expensive make_examples workers). Fixing it also should be very straightforward.

I think this is a fair price to pay in order to avoid unwanted significant extra costs that might catch our users by surprise (more specifically repeating a task with non-preemptible machines when preemptible attempts failed).

Please let me know if you agree with this.

Reply:
As discussed offline, I modified the code to make it backward compatible. We will depricate both flags in next release (issue #22). Meanwhile, we allow our users to start using the new `--attempts` flag.

Underlying logic is also fixed so when `--preemptible` is requested we only retry failures using preemptible machines.